### DISCUSSION TITLE
Unknown column 'oc_t_user.pk_i_id' in 'where clause'

### DISCUSSION THREAD

**Discussion by patriksh** [mindstellar/shopclass#469] [Q&A] [answered]
I had a Osclass 3.something website. I removed all the files, kept the database, uploaded new Osclass 5.1.2 and did the DB upgrade in admin.

Now I'm getting fatal errors at random places, wherever user ID is used for some searching. Like all user items on public profile, searching items by user in admin, etc.

I have no idea why this might be happening. Anyone has any ideas? The table and column exist obviously.

```
ERROR: Unknown column 'oc_t_user.pk_i_id' in 'where clause' in oc-includes/osclass/classes/database/DBCommandClass.php on line no 973 Error Code:1054 with context: 
'#0 oc-includes/osclass/classes/database/DBCommandClass.php(973): mysqli->query()
#1 oc-includes/osclass/classes/database/DBCommandClass.php(878): DBCommandClass->execute()
#2 oc-includes/osclass/classes/model/Search.php(579): DBCommandClass->query()
#3 oc-includes/osclass/classes/datatables/ItemsDataTable.php(63): Search->doSearch()
#4 oc-includes/osclass/classes/controller/admin/CAdminItems.php(914): ItemsDataTable->table()
#5 oc-admin/index.php(63): CAdminItems->doModel()
#6 {main}'
```

Server version: 10.4.22-MariaDB MariaDB Server
PHP 8.1.1 (cli) (built: Dec 31 2021 18:10:26) (NTS)

**Comment by navjottomer** [comment]
Run a repair on table.

**Reply by patriksh** [reply]
Cannot repair InnoDB tables, should I convert them to MyISAM?

**Comment by navjottomer** [comment]
Run a Alter query

**Reply by patriksh** [reply]
Alter what? MyISAM shouts at me about foreign key errors when I try to convert.

**Comment by navjottomer** [comment]
Run query like this it will recreate the same table.
ALTER TABLE `table_name` ENGINE = InnoDB;

**Reply by patriksh** [reply]
Did it now on item and user tables, still same errors. :/

**Reply by navjottomer** [reply]
Without looking at DB structure it's hard to identify the cause.

**Reply by patriksh** [reply]
If you want I can send you some images (tell what you need), or even give you phpMyAdmin login.
But I don't wanna waste your time... Another option is to do a clean install and then import users and ads manually.

**Comment by navjottomer** [comment]
Send the phpmyadmin credentials, let's see if that helps.

**Reply by patriksh** [reply]
Sent to your gmail.

**Reply by navjottomer** [reply]
In error you have `oc_t_user` while in database you have a different prefix, did you changed that before posting the error here?

**Reply by navjottomer** [reply]
Also make sure you're using the recommended php version 7.0 - 8.0

**Reply by patriksh** [reply]
Changed the prefix when posting here.

I'm on 8.1 now, you think it's worth trying to downgrade to 8.0?

**Comment by navjottomer** [comment, marked as answer]
Yup sure try that.

**Reply by patriksh** [reply]
Wow that actually made it work.

**Reply by patriksh** [reply]
Thanks a lot man! Now gotta figure what breaks it in 8.1 🤔

**Comment by navjottomer** [comment]
Sure some escaping issues

**Reply by patriksh** [reply]
From PHP 8.1 changelog

- HTML entity en/decode functions process single quotes and substitute by default.
- MySQLi: Default error mode set to exceptions.
